cohortextractor/query_language.py

```diff
@@ -19,8 +16,47 @@ def table(name):
     return Table(name)
 
 
-def category_group(source, comparison, value):
-    return CategoryGroup(source, _OPERATOR_MAPPING[comparison], value)
+class Comparator:
+    """A generic comparator to represent a comparison between a source object and a value"""
```

Comment:
Based on a very simplified version of django `Q` 
https://github.com/django/django/blob/main/django/db/models/query_utils.py#L29